### PR TITLE
test: refactor tests

### PR DESCRIPTION
**Pull request by michen00**
We had some redundant tests and a couple edge cases are better handled now.

---

> [!NOTE]
> Fixes toggle behavior and tightens tests.
> 
> - **Core logic**: `toggleSquigglesCore` now detects the "invisible" state via presence of `invisibleSquiggles.originalColors` and restores ALL stored colors regardless of current hide flags. Also clears stale transparent keys not in stored colors, adds `hasStoredColors`/`isInvisibleState`, and avoids no-op when nothing to apply.
> - **VS Code UX**: Status bar messaging still conditional; no functional API changes.
> - **Tests**: Remove redundant E2E status bar suite; simplify command palette E2E. Add integration test for `showStatusBarMessage`. Expand unit tests with bug scenarios (all flags off while invisible, restore across config changes) and add mocha root hooks to reset mock config state.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 0878c04a8915fd7acdb068b1a285705b18df5680. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -166,30 +178,23 @@ export function toggleSquigglesCore(
...
   const newCustomizations = { ...currentCustomizations };
 
-  if (isAlreadyTransparent) {
-    // Restore previous colors (if any), otherwise clear the customizations explicitly.
-    Object.keys(transparentColorsToApply).forEach((key) => {
+  if (isInvisibleState) {
+    // Restore ALL colors from storedColors, regardless of current checkbox settings
+    Object.keys(storedColors).forEach((key) => {
       const storedValue = (storedColors as Record<string, unknown>)[key];
       if (typeof storedValue === "string") {
```
[the remark this answers is not in the document]
Comment (michen00) [reply src/extension.ts]: [9809f6e](https://github.com/michen00/invisible-squiggles/pull/79/commits/9809f6e536a4885aad7775fdaef2e54795c5b7de) & [5e9797a](https://github.com/michen00/invisible-squiggles/pull/79/commits/5e9797aade713e00fc249012abb27a6ae38e421f)
